google-api-client/src/main/java/com/google/api/client/googleapis/javanet/GoogleNetHttpTransport.java

```diff
@@ -51,10 +58,54 @@ static HttpTransport newProxyTransport() throws GeneralSecurityException, IOExce
    */
   public static NetHttpTransport newTrustedTransport()
       throws GeneralSecurityException, IOException {
-    return new NetHttpTransport.Builder().trustCertificates(GoogleUtils.getCertificateTrustStore())
-        .build();
+    return newTrustedTransport(null);
   }
 
-  private GoogleNetHttpTransport() {
+  /**
+   * {@link Beta} <br>
+   * Returns a new instance of {@link NetHttpTransport} that uses
+   * {@link GoogleUtils#getCertificateTrustStore()} for the trusted certificates using
+   * {@link com.google.api.client.http.javanet.NetHttpTransport.Builder#trustCertificates(KeyStore)}
+   * . If `GOOGLE_API_USE_CLIENT_CERTIFICATE` environment variable is set to "true",
+   * the function looks for user provided client certificate first from 
+   * clientCertificateSource InputStream, if not exists, then the default from
+   * {@link Utils#loadDefaultCertificate()}. If client certificate exists,
+   * the transport uses it and is mutual TLS.
+   * 
+   * @param clientCertificateSource InputStream for mutual TLS client certificate and private key   
+   */
+  @Beta
+  public static NetHttpTransport newTrustedTransport(InputStream clientCertificateSource)
```

Comment:
Same comment as the ApacheTransport: we probably don't need to make this public, or alternatively provide the KeyStore/password.

Reply:
now using keystore/password